server/gendb/table.go.tmpl

```diff
@@ -236,15 +236,23 @@ func (t {{ $modelName }}Table) scan{{ $modelName }}s(ctx context.Context, input
     ConsistentRead: aws.Bool(!input.DisableConsistentRead),
   }
   if input.StartingAfter != nil {
+  {{- if indexContainsNonCompositeAttribute $xdbConfig $xdbConfig.DynamoDB.KeySchema }}
     exclusiveStartKey, err := dynamodbattribute.MarshalMap(input.StartingAfter)
     if err != nil {
       return fmt.Errorf("error encoding exclusive start key for scan: %%s", err.Error())
     }
+  {{- end }}
```

Comment:
Without this condition, `exclusiveStartKey` is unused when the index consists only of composite attributes, and the Go compiler loudly complains about that.